runtime/runtime-params-estimator/src/cost.rs

```diff
@@ -79,10 +79,24 @@ pub enum Cost {
     /// a transaction. Subtract base costs and apply least-squares on the
     /// results to find the per-byte costs.
     ActionDeployContractPerByte,
+    /// Estimates `action_creation_config.function_call_cost`, which is the base
+    /// cost for adding a `FunctionCallAction` to a receipt. It aims to account
+    /// for all costs of calling a function that are already known on the caller
+    /// side.
```

Comment:
Let's rephrase this: I *think* this is not so much about knowing, as about where the actual work is being done. Even if we know the size of the contract when making a call, we'd still like to charge for that on exec (which might be different shard, etc)

Reply:
`FunctionCallAction` does have a send and an execution part though. I would argue all at this point known cost should be part of the receipt `execution`. Using VM costs has the downside that it relies on user-attached gas which is less robust against parameter value changes.